### ISSUE TITLE
Task - Geographic Area Lookup, Geographic Area, Filter CE/CO

### ISSUE THREAD

**Issue by AntWeb-org** [SpeciesFileGroup/taxonworks#3268]
### Feature or enhancement

Using the GAL task, Data - Supporting - GA page or Filtering CEs/COs, I must know the ASCII codes for diacritics and special characters or no results are given. For example, how would I search for Île-de-France if I do not currently know the ASCII code or if I do not know that there are dashes involved in the GA? I can do a partial search in the GA page or Filter task to find all 'San José's by typing 'san jos', but if the diacritic is the first character or even second I will either not get results or have to sift through a number of results.

Also, the GAL, GA and Filters do not support the ASCII codes I _do_ know for diacritics. See screenshot for example of typing “San José” using the alt+130 ascii code in GAL and GA. 
It previously did not work, then seemed to work for a minute. Now it does not work again, so it might be an issue on my end.

@bpescador, @chrisgrinter, @rich-keller

### Location

Task - Geographic Area Lookup, Geographic Area, and CE/CO Filters

### Screenshot, napkin sketch of interface, or conceptual description

![image](https://user-images.githubusercontent.com/74991451/210465613-72ae198c-e44c-414a-8a99-5886db57a971.png)

![image](https://user-images.githubusercontent.com/74991451/210465588-d6d30e5b-e7c6-4117-919f-14dc6d0dd6df.png)

**Comment by mjy** [comment]
@AntWeb-org two things I note.
1 - GAL the search is one area per line.  It must match the name at present (but see below).  It appears you are trying multiple names per line in the screenshot.
2 - You can find a Geographic Area and use the radial annotator to add alternate spellings.  These should (I will have to double check) get picked up in the _autocomplete_ searches, but I don't think they get picked up in the GAL search.  We could add that option there.

**Comment by AntWeb-org** [comment]
1. It looks like I was typing multiple names but that is what auto-filled as I was typing the ASCII code for é. Perhaps TW doesn't like my 'function' key from my laptop and is causing some sort of cache issue. I have to use Fn+Alt+130 for the ASCII character é and 'San JosSan José' is the result.
2. The radial annotator is good to know. I'll look into it.

**Comment by mjy** [comment]
The input text box is just an input, there is nothing special about it on TW side.  If you have input caching or other special features set on your browser this might affect what you type.

**Comment by AntWeb-org** [comment]
Weird. I'll have to try it on my desktop.

The alternate spelling in the radial annotator works but I still need to know the exact names to add alternate spellings. So, if I am entering a locality via a label that shows: France: Ile de France, etc., I will still need to find the correct name first. I'll probably end up on Wikipedia anyway so perhaps it will not be an issue in the long run.

**Comment by mjy** [comment]
Remember the autocomplete has lots of magic, you can use sub-strings, etc. to limit your result, there is even a way to scope by parent to get a list of possible candidates.  We have had probably over a milliion uses of selecting a GA, it seems to work very well for a vast majority of cases.

**Comment by mjy** [comment]
@debpaul not here that we need to describe "magic" patterns that are usable for each autocomplete in help docs.

**Comment by rich-keller** [comment]
> You can find a Geographic Area and use the radial annotator to add alternate spellings.

This solution puts the onus on the user to do the considerable work of identifying and adding alternate spellings. How is the user to know there are multiple spellings with different diacritics? 

One suggestion is that the system should do this work for the user by 1) stripping the stored geographic area text labels of diacritics, 2) stripping the user's input of diacritics, and then 3) matching the two stripped versions together. This should find all GAs of possible interest to the user.

**Comment by AntWeb-org** [comment]
I agree @rich-keller. Thinking about it from a workflow perspective, each time I come across an alternate spelling of a GA whether from a handwritten label, a physical label, or an excel spreadsheet of data, I would need to stop and include it in the radial annotator. It will definitely interrupt my workflow.

**Comment by mjy** [comment]
We are largely describing a the same philosophy/approach and I agree with your concerns. I also agree that the improvements you note (full text indexing related) would help.  Where we differ (if we do) is perhaps how our workflows play out.

Let's step back and describe the logic that TW is using.

For (new) assignment/digitization:

1) Whenever possible _assert_ a spatial circumscription for your CE on top of verbatim values.

This has the following consequences:
* You can compute, spatially, against controlled vocabularies to _infer_ the labels for that CE.
* When controlled vocabularies change (e.g. a new gazetteer created by someone else) you don't match labels (diacritics, abbreviations etc. are hard), you match spatially (by finding the intersection of your shape with their shape). You use the spatial match to generate the labels.  
* We use the above strategy with TW point georeferences and the DwC index.  We automatically populate state and county labels in the DwC index by computing spatially. 
* This strategy, for POINT georeference, almost completely eliminates (in my preferred workflow, yours may vary) any need to parse any country,state,county strings.
* You can search verbatim labels with wildcards. This will let you match what you see, with what you type.  If you're familiar we've experimented with using Regex (People uniquify), and could add it to CE labels as a feature far easier than building out a full-text index on individual fields.

2) Use an autocomplete.

* I'm repeating myself here, but in our experience, in most cases, you do not need to know the exact label to find the corresponding GA in practice. The autocomplete works just fine for nearly all cases.  I'm not just stating this, existing projects have added hundreds of thousands of records using this strategy, and the issue rarely comes up in support.
* But it does come up sometimes. When?  When there is a record that is _consistently_ spelled in some other ways.  This is uncommon in our experience.  In this case add the alternate spelling. That spelling lets you match to the series of specimens that you have (typically there is some pre-sorting to optimize similar labels in a digitization workflow).
* Use your pinboard.  Most likely you have a small set of troublemakers.  Add those GAs to your pinboard, and they are accessible at your fingertips, in all GA smart selectors.  Make have use of the default GA on your pinboard to make its assignment as simple as clicking the blue pin that shows up when you have assigned a default.

For filter/search/discovery:

_Note the new unified filters in development represent a massive improvement, we hope to be testing them by the end of the month.  Even without them the following has worked well. Spatial search via reference to GA is greatly improved across *all* filters, for example. _

1) Use a spatial query to narrow the result. In many cases you don't need a label.

2) Use spatial + some word on the label (you either have information in front of you, or you don't)
* This combination is extremely effective in narrowing down to a set of COs.

3) You are not the first to suggest an approach to diacritic persistence/use in the back end.  You can find other issues on this tracker that reference this issue for People's names.  As we've worked with others, and explored the requirements, we've become convinced that it would help, but not be a global fix, we've also heard that solutions that allow for identifiers, alternate spellings, etc. have helped. Language indexing is non-trivial, the pay-off to curators/users for implementing it in code is there, but the cost (for developers, e.g. see https://www.postgresql.org/docs/current/charset.html) is high, and to be open we'd love to have it, but it's not on the "priority" horizon for the core developers (PRs welcome!).

**Comment by rich-keller** [comment]
Thanks for your detailed reply @mjy. We're pondering your response and will get back soon with some comments/reactions. Didn't want you to think we were ignoring you!

**Comment by AntWeb-org** [comment]
Regarding only the section on wildcards...

"You can search verbatim labels with wildcards. This will let you match what you see, with what you type. If you're familiar we've experimented with using Regex (People uniquify), and could add it to CE labels as a feature far easier than building out a full-text index on individual fields." 

What are the wildcards to use? Are wildcards available for only the verbatim label option under Filter, by attribute, field or can they be used under Data predicates/values, Matching identifiers, Identifiers and Tags as well?

**Comment by AntWeb-org** [comment]
can use filters/custom attributes
